Sometimes Nuget debug build fails

### Type of issue

- [X] Bug
- [ ] Enhancement
- [ ] Compliance
- [ ] Question
- [ ] Help wanted

### Current Behavior

```
  Zipping directory "D:\a\1\s\Stack\Opc.Ua.Core\obj\zipnodeset2" to "D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip".
##[error]Stack\Opc.Ua.Core\Opc.Ua.Core.csproj(100,5): Error MSB3943: Failed to zip directory "D:\a\1\s\Stack\Opc.Ua.Core\obj\zipnodeset2" to file "D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip".  The process cannot access the file 'D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip' because it is being used by another process.
D:\a\1\s\Stack\Opc.Ua.Core\Opc.Ua.Core.csproj(100,5): error MSB3943: Failed to zip directory "D:\a\1\s\Stack\Opc.Ua.Core\obj\zipnodeset2" to file "D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip".  The process cannot access the file 'D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip' because it is being used by another process. [TargetFramework=net6.0]
  Zip Schema/Opc.Ua.NodeSet2.xml in obj\/zipnodeset2.
##[error]Stack\Opc.Ua.Core\Opc.Ua.Core.csproj(100,5): Error MSB3943: Failed to zip directory "D:\a\1\s\Stack\Opc.Ua.Core\obj\zipnodeset2" to file "D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip".  The process cannot access the file 'D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip' because it is being used by another process.
D:\a\1\s\Stack\Opc.Ua.Core\Opc.Ua.Core.csproj(100,5): error MSB3943: Failed to zip directory "D:\a\1\s\Stack\Opc.Ua.Core\obj\zipnodeset2" to file "D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip".  The process cannot access the file 'D:\a\1\s\Stack\Opc.Ua.Core\Schema\Opc.Ua.NodeSet2.xml.zip' because it is being used by another process.
  Zip NodeSet2 completed, obj\/zipnodeset2 removed.
```

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
- OS:
- Environment:
- Runtime:
- Nuget Version:
- Component:
- Server:
- Client:
```


### Anything else?

_No response_

--- a/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
+++ b/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
@@ -94,7 +94,6 @@
     <EmbeddedResource Include="Schema\Opc.Ua.NodeSet2.xml.zip" />
   </ItemGroup>
 
-  <!-- ensure this target runs only once for single- and multitargeted builds -->
   <PropertyGroup>
     <ZipTmp>$(BaseIntermediateOutputPath)$(Configuration)/$(TargetFramework)/zipnodeset2</ZipTmp>
     <ZipNodeSet2XML>Schema/Opc.Ua.NodeSet2.xml</ZipNodeSet2XML>

--- a/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
+++ b/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
@@ -96,12 +96,11 @@
 
   <!-- ensure this target runs only once for single- and multitargeted builds -->
   <PropertyGroup>
-    <IsCrossTargetingProject Condition="$(TargetFrameworks) != ''">true</IsCrossTargetingProject>
-    <ZipTmp>$(BaseIntermediateOutputPath)/zipnodeset2</ZipTmp>
+    <ZipTmp>$(BaseIntermediateOutputPath)/$(Configuration)/$(TargetFramework)/zipnodeset2</ZipTmp>
     <ZipNodeSet2XML>Schema/Opc.Ua.NodeSet2.xml</ZipNodeSet2XML>
   </PropertyGroup>
 
-  <Target Name="ZipNodeSet2" Condition="'$(IsCrossTargetingProject)' == '$(IsCrossTargetingBuild)'" BeforeTargets="DispatchToInnerBuilds;BeforeBuild" Inputs="$(ZipNodeSet2XML)" Outputs="$(ZipNodeSet2XML).zip">
+  <Target Name="ZipNodeSet2" BeforeTargets="BeforeBuild" Inputs="$(ZipNodeSet2XML)" Outputs="$(ZipNodeSet2XML).zip">
     <Message Text="Zip $(ZipNodeSet2XML) in $(ZipTmp)." Importance="high" />
     <Copy SourceFiles="$(ZipNodeSet2XML)" DestinationFolder="$(ZipTmp)" />
     <ZipDirectory SourceDirectory="$(ZipTmp)" DestinationFile="$(ZipNodeSet2XML).zip" Overwrite="true" />

--- a/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
+++ b/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
@@ -9,6 +9,7 @@
     <Description>OPC UA Core Class Library</Description>
     <IsPackable>true</IsPackable>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)' == 'Debug'">
@@ -93,13 +94,14 @@
     <EmbeddedResource Include="Schema\Opc.Ua.NodeSet2.xml.zip" />
   </ItemGroup>
 
+  <!-- ensure this target runs only once for single- and multitargeted builds -->
   <PropertyGroup>
+    <IsCrossTargetingProject Condition="$(TargetFrameworks) != ''">true</IsCrossTargetingProject>
     <ZipTmp>$(BaseIntermediateOutputPath)/zipnodeset2</ZipTmp>
     <ZipNodeSet2XML>Schema/Opc.Ua.NodeSet2.xml</ZipNodeSet2XML>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
 
-  <Target Name="ZipNodeSet2" BeforeTargets="PrepareForBuild" Inputs="$(ZipNodeSet2XML)" Outputs="$(ZipNodeSet2XML).zip">
+  <Target Name="ZipNodeSet2" Condition="'$(IsCrossTargetingProject)' == '$(IsCrossTargetingBuild)'" BeforeTargets="DispatchToInnerBuilds;BeforeBuild" Inputs="$(ZipNodeSet2XML)" Outputs="$(ZipNodeSet2XML).zip">
     <Message Text="Zip $(ZipNodeSet2XML) in $(ZipTmp)." Importance="high" />
     <Copy SourceFiles="$(ZipNodeSet2XML)" DestinationFolder="$(ZipTmp)" />
     <ZipDirectory SourceDirectory="$(ZipTmp)" DestinationFile="$(ZipNodeSet2XML).zip" Overwrite="true" />

--- a/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
+++ b/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
@@ -89,27 +89,31 @@
 
   <Target Name="GetPackagingOutputs" />
 
-  <!-- Produces a zipped version of the Nodeset 2 to reduce assembly size by 2.7MB ! -->
-  <ItemGroup>
-    <EmbeddedResource Include="Schema\Opc.Ua.NodeSet2.xml.zip" />
-  </ItemGroup>
-
+  <!-- Produces a zipped version of the Nodeset 2 to reduce assembly size! -->
   <PropertyGroup>
     <ZipTmp>$(BaseIntermediateOutputPath)$(Configuration)/$(TargetFramework)/zipnodeset2</ZipTmp>
-    <ZipNodeSet2XML>Schema/Opc.Ua.NodeSet2.xml</ZipNodeSet2XML>
+    <NodeSet2XML>Schema/Opc.Ua.NodeSet2.xml</NodeSet2XML>
+    <ZipNodeSet2XML>$(BaseIntermediateOutputPath)$(Configuration)/$(TargetFramework)/Opc.Ua.NodeSet2.xml.zip</ZipNodeSet2XML>
   </PropertyGroup>
 
-  <Target Name="ZipNodeSet2" BeforeTargets="BeforeBuild" Inputs="$(ZipNodeSet2XML)" Outputs="$(ZipNodeSet2XML).zip">
-    <Message Text="Zip $(ZipNodeSet2XML) in $(ZipTmp)." Importance="high" />
-    <Copy SourceFiles="$(ZipNodeSet2XML)" DestinationFolder="$(ZipTmp)" />
-    <ZipDirectory SourceDirectory="$(ZipTmp)" DestinationFile="$(ZipNodeSet2XML).zip" Overwrite="true" />
+  <!-- Embed the zipped Nodeset2 file -->
+  <ItemGroup>
+    <EmbeddedResource Include="$(ZipNodeSet2XML)" Link="$(NodeSet2XML).zip" />
+  </ItemGroup>
+
+  <!-- Zip the NodeSet2 file -->
+  <Target Name="ZipNodeSet2" BeforeTargets="PrepareForBuild" Inputs="$(NodeSet2XML)" Outputs="$(ZipNodeSet2XML)">
+    <Message Text="Zip $(NodeSet2XML) in $(ZipNodeSet2XML)." Importance="high" />
+    <Copy SourceFiles="$(NodeSet2XML)" DestinationFolder="$(ZipTmp)" />
+    <ZipDirectory SourceDirectory="$(ZipTmp)" DestinationFile="$(ZipNodeSet2XML)" Overwrite="true" />
     <RemoveDir Directories="$(ZipTmp)" />
     <Message Text="Zip NodeSet2 completed, $(ZipTmp) removed." Importance="high" />
   </Target>
-
-  <Target Name="RemoveZipNodeset2" AfterTargets="Clean" Condition="Exists('$(ZipNodeSet2XML).zip')">
-    <Message Text="Delete Zip $(ZipNodeSet2XML).zip" Importance="high"></Message>
-    <Delete Files="$(ZipNodeSet2XML).zip"></Delete>
+  
+  <!-- Clean up the zip file -->
+  <Target Name="RemoveZipNodeset2" AfterTargets="Clean" Condition="Exists('$(ZipNodeSet2XML)')">
+    <Message Text="Delete Zip $(ZipNodeSet2XML)" Importance="high"></Message>
+    <Delete Files="$(ZipNodeSet2XML)"></Delete>
   </Target>
-
+  
 </Project>

--- a/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
+++ b/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
@@ -96,7 +96,7 @@
 
   <!-- ensure this target runs only once for single- and multitargeted builds -->
   <PropertyGroup>
-    <ZipTmp>$(BaseIntermediateOutputPath)/$(Configuration)/$(TargetFramework)/zipnodeset2</ZipTmp>
+    <ZipTmp>$(BaseIntermediateOutputPath)$(Configuration)/$(TargetFramework)/zipnodeset2</ZipTmp>
     <ZipNodeSet2XML>Schema/Opc.Ua.NodeSet2.xml</ZipNodeSet2XML>
   </PropertyGroup>
 
@@ -108,7 +108,7 @@
     <Message Text="Zip NodeSet2 completed, $(ZipTmp) removed." Importance="high" />
   </Target>
 
-  <Target Name="RemoveZipNodeset2" AfterTargets="Clean">
+  <Target Name="RemoveZipNodeset2" AfterTargets="Clean" Condition="Exists('$(ZipNodeSet2XML).zip')">
     <Message Text="Delete Zip $(ZipNodeSet2XML).zip" Importance="high"></Message>
     <Delete Files="$(ZipNodeSet2XML).zip"></Delete>
   </Target>

--- a/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
+++ b/Stack/Opc.Ua.Core/Opc.Ua.Core.csproj
@@ -108,4 +108,9 @@
     <Message Text="Zip NodeSet2 completed, $(ZipTmp) removed." Importance="high" />
   </Target>
 
+  <Target Name="RemoveZipNodeset2" AfterTargets="Clean">
+    <Message Text="Delete Zip $(ZipNodeSet2XML).zip" Importance="high"></Message>
+    <Delete Files="$(ZipNodeSet2XML).zip"></Delete>
+  </Target>
+
 </Project>

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -68,7 +68,7 @@ stages:
     parameters:
       configuration: Release
       framework: net6.0
-      agents: '@{ windows = "windows-2022"; linux="ubuntu-22.04"; mac = "macOS-12"}'
+      agents: '@{ windows = "windows-2022"; linux="ubuntu-22.04"; mac = "macOS-15"}'
       jobnamesuffix: net60
       customtestarget: net6.0
 - stage: testreleasepr

